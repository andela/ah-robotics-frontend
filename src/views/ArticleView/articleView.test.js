import React from 'react';
import { ArticleView } from './index';

let wrapper;
let instance;
let props;

describe('article view tests', () => {
  props = {
    match: { params: { slug: '' } },
    fetchSingle: jest.fn(),
    articles: {},
    history: { push: jest.fn() },
    uploadImageAction: jest.fn(),
  };
  beforeEach(() => {
    wrapper = shallow(<ArticleView {...props} />);
    instance = wrapper.instance();
  });
  it('should render article view', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should change state when onImageChange is called', () => {
    const event = {
      target: {
        name: 'image',
        files: [
          [{ name: 'test-image' }],
        ],
      },
    };

    instance.onImageChange(event);
    expect(instance.state.image).toEqual(event.target.files[0]);
  });
  it('should redirect onHandleEdit', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    instance.handleEdit(event);
    expect(instance.props.history.push.mock.calls[0]).toEqual(['/articles/update/']);
  });
});
