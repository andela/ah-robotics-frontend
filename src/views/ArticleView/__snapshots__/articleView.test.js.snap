// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article view tests should render article view 1`] = `
<div>
  <Connect(ArticleComponent)
    getInputFocus={[Function]}
    handleEdit={[Function]}
    imageInput={
      Object {
        "current": null,
      }
    }
    onImageChange={[Function]}
    rating={<withRouter(Connect(RatingsView)) />}
  />
</div>
`;
